components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -142,6 +144,12 @@ public int createApplication(ServiceProvider serviceProvider, String tenantDomai
         } finally {
             endTenantFlow();
         }
+
+        for (ApplicationMgtListener listener : listeners) {
+            if (!listener.doPostCreateApplication(serviceProvider)) {
```

Comment:
At this point we have already ended the tenant flow and no way of knowing the tenant domain or username that was passed to the service method.